src/api/handleIncomingSubmission.ts

```diff
@@ -24,63 +23,38 @@ import { log } from '../utils/logger'
  *   - Register the submission for voting via submissions manager and persist it in the DB
  */
 export default async function handler (apiSubmission: IncomingSubmissionData): Promise<Result<unknown, Error>> {
-  // Before proceeding further, register with the DB. This cannot fail, if it does, we abort.
-  const idRes = await submissions.generateId()
-
-  if (idRes.err) {
-    return Err(idRes.val)
+  // Before proceeding further, register with the DB.
+  const submissionRes = await submissions.createProject(apiSubmission)
+
+  if (submissionRes.err) {
+    // If the database throws an error, we need to abort
+    // Unfortunately project data will have to be lost
+    await reportError('Database threw an error while attempting to register the project', { name: apiSubmission.name, shouldRelate: false })
+    return Err(submissionRes.val)
   }
 
-  const id = idRes.val
-
-  const submission = {
-    id,
-    relatedMessages: [] as RelatedMessage[],
-    votes: [] as Vote[],
-    drafts: [] as Draft[],
-    links: [
-      {
-        submissionId: id,
-        submissionDetailsId: id,
-        type: 'SOURCE',
-        url: apiSubmission.links.source
-      },
-      {
-        submissionId: id,
-        submissionDetailsId: id,
-        type: 'OTHER',
-        url: apiSubmission.links.other
-      }
-    ] as SubmissionLink[],
-
-    details: {
-      name: apiSubmission.name,
-      authorId: apiSubmission.author,
-      description: apiSubmission.description,
-      techUsed: apiSubmission.tech,
-      state: 'PROCESSING' as const,
-
-      // IDs, to be set,
-      messageId: '',
-      reviewThreadId: '',
-      submissionId: id
-    }
-
-  }
+  const submission = submissionRes.val
 
   // Fetch project author
   const authorRes = await Result.wrapAsync(async () => await client.users.fetch(apiSubmission.author))
 
   if (authorRes.err) {
-    const err = authorRes.val
+    const err = authorRes.val as DiscordAPIError
 
     log.error(`Could not fetch author for submission ${apiSubmission.name}`)
     log.error(err)
 
-    // TODO: reject the project here and cease processing
+    // Reject the project if the user id is not found or the snowflake provided is invalid
+    if (err.code === 404 || err.code === 400) {
+      // @ts-expect-error client user will not be null
```

Comment:
```suggestion
      assert(client.user, 'client.user was not set')
```
using our assertion function.